src/app/states/ChooseNPCMenuState.cpp

```diff
@@ -0,0 +1,372 @@
+    arrowHovered[0] = false;
+    arrowHovered[1] = false;
+}
+
+void ChooseNPCGameState::handleInput()
+{
+    // Navegación con flechas horizontales
+    if (IsKeyPressed(KEY_RIGHT) || IsKeyPressed(KEY_D))
+    {
+        currentCharacterIndex = (currentCharacterIndex + 1) % characters.size();
+    }
+
+    if (IsKeyPressed(KEY_LEFT) || IsKeyPressed(KEY_A))
+    {
+        currentCharacterIndex = (currentCharacterIndex - 1 + characters.size()) % characters.size();
+    }
+
+    // Confirmar selección
+    if (IsKeyPressed(KEY_SPACE) || IsKeyPressed(KEY_ENTER))
+    {
+        state_machine->add_state(std::make_unique<MainGameState>(characters[currentCharacterIndex].type), true);
+    }
+
+    // Volver al menú anterior (reemplazar estado actual con MainMenu)
+    if (IsKeyPressed(KEY_Q))
+    {
+        state_machine->add_state(std::make_unique<MainMenuState>(), true);
+    }
+
+    // Detectar hover del ratón sobre los botones de flecha
+    Vector2 mousePosition = GetMousePosition();
+    int screenWidth = GetScreenWidth();
+    int screenHeight = GetScreenHeight();
+
+    // Calcular posiciones para los botones
+    Vector2 containerPos = GetContainerPos(screenWidth, screenHeight);
+    float spriteWidth = CONTAINER_SIZE * CONTAINER_SCALE;
+    float spriteHeight = spriteWidth;
+    Vector2 spritePos = GetSpritePos(containerPos, spriteWidth, spriteHeight);
+
+    // Resetear el array de hover
+    arrowHovered[0] = false;
+    arrowHovered[1] = false;
+
+    // Verificar hover sobre cada botón
+    for (int i = 0; i < 2; i++)
+    {
+        Rectangle btn = GetArrowButtonRect(i, spritePos, spriteWidth, screenHeight);
+        if (CheckCollisionPointRec(mousePosition, btn))
+        {
+            arrowHovered[i] = true;
+
+            // Si se hace clic, navegar
+            if (IsMouseButtonPressed(MOUSE_LEFT_BUTTON))
+            {
+                if (i == 0) // Flecha izquierda
+                {
+                    currentCharacterIndex = (currentCharacterIndex - 1 + characters.size()) % characters.size();
+                }
+                else // Flecha derecha
+                {
+                    currentCharacterIndex = (currentCharacterIndex + 1) % characters.size();
+                }
+            }
+        }
+    }
+
+    // Verificar clic en el recuadro del personaje para seleccionarlo
+    Rectangle containerRect = {containerPos.x, containerPos.y, DISPLAY_SIZE, DISPLAY_SIZE};
+
+    if (CheckCollisionPointRec(mousePosition, containerRect))
+    {
+        if (IsMouseButtonPressed(MOUSE_LEFT_BUTTON))
+        {
+            // Confirmar selección del personaje actual
+            state_machine->add_state(std::make_unique<MainGameState>(characters[currentCharacterIndex].type), true);
+        }
+    }
+}
+
+void ChooseNPCGameState::update(float deltaTime __attribute__((unused)))
+{
+    // Por ahora no hay lógica de actualización
+    // Aquí se podría agregar animación de sprites si se desea
+}
```

Comment:
```suggestion
void ChooseNPCGameState::update(float deltaTime)
{

}
```
Ya que tienes que cambiar al estándar de carga de imágenes de la aplicación aprovecha y mete aquí en el update las animaciones, total solo tienes que copiar y pegar lo de Player